File: production/shared/src/main/scala/kpn/shared/common/TrackPath.scala
package kpn.shared.common

case class TrackPath(
  startNodeId: Long,
  endNodeId: Long,
  meters: Int,
  segments: Seq[TrackSegment]
) {
  override def toString: String = ToStringBuilder(this.getClass.getSimpleName).
    field("startNodeId", startNodeId).
    field("endNodeId", endNodeId).
    field("meters", meters).
    field("segments", segments).
    build
}

File: planner-prototype/server/src/main/java/kpn/planner/domain/document/TrackPath.java
package kpn.planner.domain.document;

import java.util.List;

public class TrackPath {

	private Long startNodeId;
	private Long endNodeId;
	private Long meters;
	private List<TrackSegment> segments;

	public TrackPath() {
	}

	public Long getStartNodeId() {
		return startNodeId;
	}

	public void setStartNodeId(Long startNodeId) {
		this.startNodeId = startNodeId;
	}

	public Long getEndNodeId() {
		return endNodeId;
	}

	public void setEndNodeId(Long endNodeId) {
		this.endNodeId = endNodeId;
	}

	public Long getMeters() {
		return meters;
	}

	public void setMeters(Long meters) {
		this.meters = meters;
	}

	public List<TrackSegment> getSegments() {
		return segments;
	}

	public void setSegments(List<TrackSegment> segments) {
		this.segments = segments;
	}
}
